From 3904312ad529661a602c5f189427e99535b709fc Mon Sep 17 00:00:00 2001
From: Steve Kondik <shade@chemlab.org>
Date: Mon, 23 Dec 2013 22:58:07 -0800
Subject: [PATCH 016/171] camera: Check if video sizes are available

 * Some drivers don't report video sizes and we should avoid using fine
   resolution lookup, otherwise a crash results.

 * Just honor the defined camcorder profiles.

Change-Id: I83e8fe2ca9bb489332f37767906bc7c56389f949
---
 src/com/android/camera/CameraSettings.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/camera/CameraSettings.java b/src/com/android/camera/CameraSettings.java
index 51ff15a90..563a9da3d 100755
--- a/src/com/android/camera/CameraSettings.java
+++ b/src/com/android/camera/CameraSettings.java
@@ -1385,6 +1385,10 @@ public class CameraSettings {
     }
     private static boolean checkSupportedVideoQuality(Parameters parameters,int width, int height){
         List <Size> supported = parameters.getSupportedVideoSizes();
+        if (supported == null) {
+            // video-size not specified in parameter list. just go along with the profile.
+            return true;
+        }
         int flag = 0;
         for (Size size : supported){
             //since we are having two profiles with same height, we are checking with height
-- 
2.17.1

